src/routes/gis/index.ts

```diff
@@ -19,84 +17,139 @@ const router = express.Router();
  * @returns GeoJSON feature collection
  */
 router.get('/feature', async (req, res) => {
+  const records = [];
   try {
-    const property = {
-      Polygon: {
-        type: 'Polygon',
-        generateProperties: () => {
-          const randomString = Math.random().toString(36).substring(7);
-          return {
-            name: `Polygon ${randomString}`,
-          };
-        },
-        numGeometries: 1000,
-      },
-      LineString: {
-        type: 'LineString',
-        generateProperties: () => {
-          const randomString = Math.random().toString(36).substring(7);
-          return {
-            name: `LineString ${randomString}`,
-          };
-        },
-        numGeometries: 1000,
-      },
-      Point: {
-        type: 'Point',
-        generateProperties: () => {
-          const randomString = Math.random().toString(36).substring(7);
-          return {
-            name: `Point ${randomString}`,
-          };
+    const layer = await Layer.findOne({ _id: req.query.id });
+    if (!layer) {
+      return res.status(404).send(i18next.t('common.errors.dataNotFound'));
+    }
+    const resourceData = await Resource.findOne({
+      layouts: {
+        $elemMatch: {
+          _id: layer.datasource.layout,
         },
-        numGeometries: 1000,
       },
-    };
-    const geoType: any = req.query.type;
-    const geoJsonData = generateGeoJson(property[geoType]);
-
-    console.log(
-      'befor GeoJson simplify size in bytes ===>>> ',
-      getGeoJsonSize(geoJsonData, 'Bytes')
-    );
+    });
 
-    /**
-     * Simplify Polygon and LineString geo json data
-     */
-    const tolerance: any = req.query.tolerance ? req.query.tolerance : 1;
-    const highQuality: any = req.query.highquality
-      ? req.query.highquality
-      : true;
-    let features: any;
-    switch (geoType) {
-      case 'Point':
-        features = generateProperties(geoJsonData, property[geoType]);
-        break;
-      case 'Polygon':
-      case 'LineString':
-        const simplifiedGeoJson = turf.simplify(geoJsonData, {
-          tolerance: tolerance,
-          highQuality: highQuality,
-        });
-        console.log(
-          'after GeoJson simplify size in bytes ===>>> ',
-          getGeoJsonSize(simplifiedGeoJson, 'Bytes')
-        );
-        features = generateProperties(simplifiedGeoJson, property[geoType]);
-        break;
+    if (
+      !resourceData ||
+      !resourceData.layouts ||
```

Comment:
@ketanreliefapps that should work for the layout so
let's try to make it work for aggregation also

Reply:
Done